### PR TITLE
Add transactions BBE test cases

### PR DESCRIPTION
**Pull request by rdhananjaya**
## Purpose
Add transactions BBE test cases

### CODE REVIEW

File: examples/transactions-distributed/tests/xa_transactions_test.bal [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    moduleName: "ballerina/io",
+    functionName: "println"
+}
+public function mockPrint(any... s) {
+    outputs[count] = string.convert(s[0]);
+    count += 1;
+}
+
+@test:Config
+function testFunc() {
```
Comment (hasithaa) [inline examples/transactions-distributed/tests/xa_transactions_test.bal]: I can't see this output in ballerina-lang/examples/transactions-distributed/participant.out file. Is this correct?

Comment (rdhananjaya) [reply examples/transactions-distributed/tests/xa_transactions_test.bal]: This file was mistakenly committed, we were looking at a way to test distributed scenario, specially the interleaving log/print entries.

File: examples/transactions-distributed/initiator.bal [hunk 2]
```diff
@@ -15,14 +15,13 @@ service InitiatorService on new http:Listener(8080) {
     }
     resource function init(http:Caller conn, http:Request req) {
         http:Response res = new;
-        log:printInfo("Initiating transaction...");
+        io:println("Initiating transaction...");
```
[the remark this answers is not in the document]
Comment (rdhananjaya) [reply examples/transactions-distributed/initiator.bal]: I was not aware of such restriction on BBE.

[the remark this answers is not in the document]
Comment (rdhananjaya) [reply examples/transactions-distributed/initiator.bal]: Fixed

**Pull request merged**